DatadogSessionReplay/Sources/Recorder/RecordingCoordinator.swift

```diff
@@ -18,11 +20,17 @@ internal class RecordingCoordinator {
     let srContextPublisher: SRContextPublisher
 
     private var currentRUMContext: RUMContext? = nil
+    @ReadWriteLock
     private var isSampled = false
 
+    /// `recordingEnabled` is used to track when the user 
+    /// has enabled or disabled the recording for Session Replay.
+    @ReadWriteLock
+    private(set) var recordingEnabled = false
```

Comment:
/suggestion Instead of using locks here (I know I made the suggestion 😄), we could use the scheduler's queue to synchronise updated. WDYT?

Reply:
So, something like this? 

```
private var _isRecordingPaused = true
private var isRecordingPaused: Bool {
    get {
            scheduler.queue.sync { _isRecordingPaused }
    }
    set {
            scheduler.queue.sync { _isRecordingPaused = newValue }
    }
}
```